### ISSUE TITLE
Unable to locate phpcs (again and again and again)

### ISSUE THREAD

**Issue by briantully** [ikappas/vscode-phpcs#84]
Hello,
I tried to install the phpcs plugin for Visual Studio Code but I keep getting phpcs: Unable to locate phpcs. Please add phpcs to your global path or use composer depency manager to install it in your project locally. message. I can access phpcs from the Command Line, it is available globally in the PATH.

phpcs is on my global PATH and was installed previously via a global composer install. It is located at ~/.composer/vendor/bin/phpcs

The call to phpcs from Command Line is following:
```
$ phpcs --version
PHP_CodeSniffer version 2.9.1 (stable) by Squiz (http://www.squiz.net)
```
I've tried other workarounds by installing via pear and homebrew, but it seems my composer installed phpcs takes priority (since it appears in my PATH first). Plus I already have other editors (PHPStorm, Atom) configured to use the composer installed phpcs with additional coding standards.

Why can't the vscode-phpcs extension allow a config setting to manually specify the location of phpcs???

**Comment by ADmad** [comment]
> Why can't the vscode-phpcs extension allow a config setting to manually specify the location of phpcs???

The ability to specify phpcs location has been implemented on the `develop` branch and should be available in next stable release.

Sadly it seems @ikappas doesn't have enough time to maintain this plugin, so no idea when `1.0.0` stable would be released.

**Comment by ADmad** [comment]
In the meantime you can check out the [phpcbf](https://marketplace.visualstudio.com/items?itemName=persoderlind.vscode-phpcbf) extension which allows you to auto format/fix possible errors using your phpcs rules.

**Comment by jawittdesigns** [comment]
> The ability to specify phpcs location has been implemented on the develop branch and should be available in next stable release.

This is off topic, but any ETA on the next release? This is the one thing that's stopping me from using vscode (I really want to). I work a lot with PHP and use custom sniffers so setting up phpcs for each project isn't an option for me.

**Comment by ikappas** [comment]
I just pushed multi-root support in develop branch that needs testing

**Comment by ikappas** [comment]
I just released vscode-phpcs 1.0.0 which supports the VSCode 1.20+ and includes many enhancements most notable of which are:

1. multi root support
2. lint as you type support
3. automatic configuration search support
4. ability to set phpcs.executablePath
5. ability to set phpcs.showWarnings
7. ability to set phpcs.showSources
8. ability to set phpcs.composerJsonPath to something other that workspaceRoot/composer.json
9. ability to enable phpcs.trace.server with messages or verbose option for tracing errors.

I would like to give my thanks to all the people that helped track down issues in the previous betas and hope the new features help you get things done easier.

On that note, I am closing this issue to properly track any new issues in the new release. If you find this issue persists or any other issue please feel free to open a new ticket for v.1.0.0.
